core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -155,7 +155,7 @@ public static String toJson(TableMetadata metadata) {
     }
   }
 
-  private static void toJson(TableMetadata metadata, JsonGenerator generator) throws IOException {
```

Comment:
I don't think we want to change accessibility on these methods, why is the custom JsonGenerator needed in the Nessie code?